src/lib/expression/expression_utils.hpp

```diff
@@ -116,8 +117,14 @@ bool expression_evaluable_on_lqp(const std::shared_ptr<AbstractExpression>& expr
 /**
  * Convert "a AND b AND c" to [a,b,c] where a,b,c can be arbitrarily complex expressions
  */
-std::vector<std::shared_ptr<AbstractExpression>> expression_flatten_conjunction(
-    const std::shared_ptr<AbstractExpression>& expression);
+std::vector<std::shared_ptr<AbstractExpression>> flatten_logical_expressions(
+    const std::shared_ptr<AbstractExpression>& expression, const LogicalOperator logical_operator);
+
+/**
+ * Convert [a,b,c], AND into "a AND b AND c"
```

Comment:
(a AND b) AND c? Above as well?

Reply:
I see no value in that here, either, but have changed it anyway